tests/test_sinogram.py

```diff
@@ -0,0 +1,138 @@
+def dtype(request):
+    """
+    Dtypes for image.
+    """
+    return request.param
+
+
+@pytest.fixture(params=IMG_SIZES, ids=lambda x: f"px={x}", scope="module")
+def img_size(request):
+    """
+    Image size.
+    """
+    return request.param
+
+
+@pytest.fixture(params=ANGLES, ids=lambda x: f"angles={x}", scope="module")
+def num_ang(request):
+    """
+    Angles (Degrees).
+    """
+    return request.param
+
+
+@pytest.fixture
+def masked_image(dtype, img_size):
+    """
+    Creates a masked image fixture using camera data from Scikit-Image.
+    """
+    g = grid_2d(img_size, normalized=True, shifted=True)
+    mask = g["r"] < 1
+
+    image = data.camera().astype(dtype)
+    image = image[:img_size, :img_size]
+    return Image(image * mask)
+
+
+# Image.project and compare results to skimage.radon
+def test_image_project(masked_image, num_ang):
+    """
+    Test Image.project on a single stack of images. Compares project method output with skimage project.
+    """
+    ny = masked_image.resolution
+    angles = np.linspace(0, 360, num_ang, endpoint=False)
+    rads = angles / 180 * np.pi
+    s = masked_image.project(rads)
+    assert s.shape == (1, len(angles), ny)
+
+    # sci-kit image `radon` reference
+    #
+    # Note, Image.project's angles are wrt projection line (ie
+    # grid), while sk's radon are wrt the image. To correspond the
+    # rotations are inverted.  This was the convention prefered by
+    # the original author of this method.
+    #
+    # Note, transpose sk output to match (angles, points)
+    reference_sinogram = radon(masked_image._data[0], theta=angles[::-1]).T
+    assert reference_sinogram.shape == (len(angles), ny), "Incorrect Shape"
+
+    # compare project method on ski-image reference
+    nrms = np.sqrt(np.mean((s[0] - reference_sinogram) ** 2, axis=-1)) / np.linalg.norm(
+        reference_sinogram, axis=-1
+    )
+
+    np.testing.assert_array_less(nrms, SK_TOL, "Error in image projections.")
+
+
+def test_multidim(num_ang):
+    """
+    Test Image.project on stacks of images. Extension of test_image_project but for multi-dimensional stacks.
+    """
+
+    L = 512  # pixels
+    n = 3
+    m = 2
+
+    # Generate a mask
+    g = grid_2d(L, normalized=True, shifted=True)
+    mask = g["r"] < 1
+
+    # Generate images
+    imgs = Image(np.random.random((m, n, L, L))) * mask
+
+    # Generate line project angles
+    angles = np.linspace(0, 360, num_ang, endpoint=False)
+    rads = angles / 180.0 * np.pi
+    s = imgs.project(rads)
+
+    # Compare
+    reference_sinograms = np.empty((m, n, num_ang, L))
+    for i in range(m):
+        for j in range(n):
+            img = imgs[i, j]
+            # Compute the singleton case, and compare with stack.
+            single_sinogram = img.project(rads)
+
+            # These should be allclose up to determinism in the FFT and NUFFT.
+            np.testing.assert_allclose(s[i, j : j + 1], single_sinogram)
```

Comment:
Isn't `s[i, j : j + 1]` just `s[i, j]`?

Reply:
nope  , try it out 😇

Reply:
Oops, my bad. Forgot that `testing.assert_allclose` is slighlty different that `allclose` in that it checks shapes.